src/Compilers/CSharp/Test/Semantic/Semantics/StructConstructorTests.cs

```diff
@@ -0,0 +1,1669 @@
+                Diagnostic(ErrorCode.ERR_NonPublicParameterlessStructConstructor, "E2").WithLocation(25, 43),
+                // (26,35): error CS8918: The parameterless struct constructor must be 'public'.
+                //     private protected struct E3 { E3() { } }
+                Diagnostic(ErrorCode.ERR_NonPublicParameterlessStructConstructor, "E3").WithLocation(26, 35),
+                // (29,34): error CS8918: The parameterless struct constructor must be 'public'.
+                //     private struct F1 { internal F1() { } }
+                Diagnostic(ErrorCode.ERR_NonPublicParameterlessStructConstructor, "F1").WithLocation(29, 34),
+                // (30,33): error CS8918: The parameterless struct constructor must be 'public'.
+                //     private struct F2 { private F2() { } }
+                Diagnostic(ErrorCode.ERR_NonPublicParameterlessStructConstructor, "F2").WithLocation(30, 33),
+                // (31,25): error CS8918: The parameterless struct constructor must be 'public'.
+                //     private struct F3 { F3() { } }
+                Diagnostic(ErrorCode.ERR_NonPublicParameterlessStructConstructor, "F3").WithLocation(31, 25));
+        }
+
+        [InlineData("assembly")]
+        [InlineData("private")]
+        [Theory]
+        public void NonPublicParameterlessConstructor_02(string accessibility)
+        {
+            var sourceA =
+$@".class public sealed S extends [mscorlib]System.ValueType
+{{
+    .method {accessibility} hidebysig specialname rtspecialname instance void .ctor() cil managed {{ ret }}
+}}";
+            var refA = CompileIL(sourceA);
+
+            var sourceB =
+@"using System;
+class Program
+{
+    static T CreateNew<T>() where T : new() => new T();
+    static T CreateStruct1<T>() where T : struct => new T();
+    static T CreateStruct2<T>() where T : struct => CreateNew<T>();
+    static string Invoke(Func<object> f)
+    {
+        object obj;
+        try
+        {
+            obj = f();
+        }
+        catch (Exception e)
+        {
+            obj = e;
+        }
+        return obj.GetType().FullName;
+    }
+    static void Main()
+    {
+        Console.WriteLine(Invoke(() => new S()));
+        Console.WriteLine(Invoke(() => CreateNew<S>()));
+        Console.WriteLine(Invoke(() => CreateStruct1<S>()));
+        Console.WriteLine(Invoke(() => CreateStruct2<S>()));
+    }
+}";
+            CompileAndVerify(sourceB, references: new[] { refA }, expectedOutput:
+@"S
+System.MissingMethodException
+System.MissingMethodException
+System.MissingMethodException");
+        }
+
+        [InlineData("internal")]
+        [InlineData("private")]
+        [Theory]
+        public void NonPublicParameterlessConstructor_03(string accessibility)
+        {
+            var sourceA =
+$@"public struct S
+{{
+    {accessibility} S() {{ }}
+}}";
+            var comp = CreateCompilation(sourceA, parseOptions: TestOptions.RegularPreview);
```

Comment:
I thought this was supposed to be a compiler error?

Reply:
This is an error but the error was not being verified since the focus of the test was using the resulting type. I've added `VerifyDiagnostics()` so the expected behavior is clear.